### PR TITLE
Added CloudTrail bucket security: encryption, logging, TLS enforcement, and versioning

### PR DESCRIPTION
**Pull request by alannix-lw**
First go-round on trying to implement AWS best practices for S3 buckets that get created by the provider.

### CODE REVIEW

File: aws/modules/cloudtrail/main.tf [hunk 1]
```diff
@@ -30,10 +31,47 @@ resource "aws_s3_bucket" "cloudtrail_bucket" {
...
+	bucket        = local.log_bucket_name
+	force_destroy = var.bucket_force_destroy
+	acl           = "log-delivery-write"
+
+	versioning {
+		enabled = var.bucket_enable_versioning
+	}
+
+	dynamic "server_side_encryption_configuration" {
+		for_each = var.bucket_enable_encryption == true ? [1] : []
```
Comment (afiune) [inline aws/modules/cloudtrail/main.tf:69]: can you use `count` instead?

Comment (TheAlanNix) [reply aws/modules/cloudtrail/main.tf:69]: To my knowledge, `count` cannot be used with a dynamic block.

**Comment by TheAlanNix** [comment]
Updated/tested with logic to only create the logging bucket when desired.

File: aws/modules/cloudtrail/main.tf (hunk 1 above)
Comment (afiune) [reply aws/modules/cloudtrail/main.tf:69]: True! Thanks for the confirmation 💯

**Comment by alannix-lw** [comment]
Encryption, Access Logging, Versioning are now disabled by default, and I've added the capability for the user to configure KMS encryption.

**Comment by afiune** [comment]
Great work Alan, I'll test this on our test environment and report back my findings, if everything looks good, I'll approve and merge! 🎉

**Pull request merged**